app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/rejectedgoodsmultiple/EnterClaimController.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.rejectedgoodsmultiple
+
+import play.api.mvc.Action
+import play.api.mvc.AnyContent
+import play.api.mvc.Call
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.config.ViewConfig
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.Forms
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.JourneyControllerComponents
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.journeys.RejectedGoodsMultipleJourney
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.TaxCode
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.ids.MRN
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.claims.mrn_does_not_exist
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.rejectedgoods.enter_claim
+
+import javax.inject.Inject
+import javax.inject.Singleton
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class EnterClaimController @Inject() (
+  val jcc: JourneyControllerComponents,
+  enterClaim: enter_claim,
+  mrnDoesNotExistPage: mrn_does_not_exist
+)(implicit val ec: ExecutionContext, viewConfig: ViewConfig)
+    extends RejectedGoodsMultipleJourneyBaseController {
+
+  val key: String          = "enter-claim.rejected-goods"
+  val subKey: Some[String] = Some("multiple")
+
+  val claimsSummaryAction: Call                 = routes.CheckClaimDetailsController.show()
+  val selectDutiesAction: Int => Call           = routes.SelectTaxCodesController.show(_)
+  val enterClaimAction: (Int, TaxCode) => Call  = routes.EnterClaimController.show(_, _)
+  val submitClaimAction: (Int, TaxCode) => Call = routes.EnterClaimController.submit(_, _)
+
+  def show(index: Int, taxCode: TaxCode): Action[AnyContent] = actionReadJourney { implicit request => journey =>
+    journey
+      .getNthMovementReferenceNumber(index - 1)
```

Comment:
Do we want to move the `-1` into the journey function? This hides the implementation.

Reply:
I would rather keep it 0-based in the function

Reply:
Alternatively, I have renamed values to distinguish between `pageIndex` and `mrnIndex`